FirebaseRemoteConfig/Sources/RCNFetch.m

```diff
@@ -264,20 +237,81 @@ - (void)refreshInstanceIDTokenAndFetchCheckInInfoWithCompletionHandler:
                                                    }]];
         }
 
-        FIRLogInfo(kFIRLoggerRemoteConfig, @"I-RCN000022", @"Success to get iid : %@.",
-                   strongSelfQueue->_settings.configInstanceID);
+        // We have a valid FIS token. We do not need to send a FID upstream anymore.
+        __weak RCNConfigFetch *weakSelf = self;
+        // Dispatch to the RC serial queue to update settings on the queue.
+        dispatch_async(self->_lockQueue, ^{
+          RCNConfigFetch *strongSelfQueue = weakSelf;
 
-        // Continue the fetch regardless of whether fetch of instance ID succeeded.
-        [strongSelfQueue fetchCheckinInfoWithCompletionHandler:completionHandler];
-      });
-    }];
-  };
-  FIRLogDebug(kFIRLoggerRemoteConfig, @"I-RCN000039", @"Starting requesting token.");
-  // Note: We expect the GCMSenderID to always be available by the time this request is made.
-  [instanceID tokenWithAuthorizedEntity:_options.GCMSenderID
-                                  scope:kInstanceIDScopeConfig
-                                options:nil
-                                handler:instanceIDHandler];
+          // Update config settings with the IID and token.
+          strongSelfQueue->_settings.configInstanceIDToken = tokenResult.authToken;
+
+          FIRLogInfo(kFIRLoggerRemoteConfig, @"I-RCN000022", @"Success to get installation token");
+
+          // Continue with checkin.
+          [strongSelfQueue fetchCheckinInfoWithCompletionHandler:completionHandler];
+        });
+      }
+      //  FIRInstanceIDTokenHandler instanceIDHandler = ^(NSString *token, NSError *error) {
+      //    if (!token || error) {
+      //      NSString *errorDescription =
+      //          [NSString stringWithFormat:@"Failed to get InstanceID token. Error : %@.", error];
+      //      FIRLogError(kFIRLoggerRemoteConfig, @"I-RCN000073", @"%@",
+      //                  [NSString stringWithFormat:@"%@", errorDescription]);
+      //      self->_settings.isFetchInProgress = NO;
+      //      return [self
+      //          reportCompletionOnHandler:completionHandler
+      //                         withStatus:FIRRemoteConfigFetchStatusFailure
+      //                          withError:[NSError errorWithDomain:FIRRemoteConfigErrorDomain
+      //                                                        code:FIRRemoteConfigErrorInternalError
+      //                                                    userInfo:@{
+      //                                                      NSLocalizedDescriptionKey :
+      //                                                      errorDescription
+      //                                                    }]];
+      //    }
+
+      // If the token is available, try to get the instanceID.
+      __weak RCNConfigFetch *weakSelf = self;
+  [instanceID getIDWithHandler:^(NSString *_Nullable identity, NSError *_Nullable error) {
```

Comment:
It looks like FIS SDK API can be improved by adding `installationID` property to [FIRInstallationsAuthTokenResult](https://github.com/firebase/firebase-ios-sdk/blob/549ba364a3e692a3f486706d51b2175ac3f5fc71/FirebaseInstallations/Source/Library/Public/FIRInstallationsAuthTokenResult.h#L23). Then a client that needs both Installation ID and Auth Token will be able to call only one[authTokenWithCompletion](https://github.com/firebase/firebase-ios-sdk/blob/549ba364a3e692a3f486706d51b2175ac3f5fc71/FirebaseInstallations/Source/Library/Public/FIRInstallations.h#L93) method instead of two (one for FID another for Auth Token).